### PR TITLE
Polish around registration and information modal

### PR DESCRIPTION
**Pull request by orta**
Currently has a subset of #1170

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Error</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td><p>No CHANGELOG changes made</p>
</td>
    </tr>
    
  </tbody>
</table>
  

  
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">2 Warnings</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td><p>PR is classed as Work in Progress</p>
</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td><p>Needs testing on a Phone if change is non-trivial</p>
</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

File: Artsy/View_Controllers/ARSerifNavigationViewController.m [hunk 1]
```diff
@@ -212,4 +212,18 @@ - (void)center:(UIView *)viewToCenter
     viewToCenter.frame = newFrame;
 }
 
+- (void)hideNavigationBarShadow:(BOOL)hide
```
Comment (ashfurrow) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: It's also possible to set the `hideNavigationBarShadow` property of `UINavigationBar`.

File: Artsy/View_Controllers/ARSerifNavigationViewController.m [hunk 2]
```diff
@@ -121,13 +119,15 @@ - (void)navigationController:(UINavigationController *)navigationController will
         static CGFloat xOffset = 4;
         CGRect labelFrame = label.bounds;
         label.frame = CGRectOffset(labelFrame, xOffset, 0);
-        UIView *titleMarginWrapper = [[UIView alloc] initWithFrame:(CGRect){ CGPointZero, { CGRectGetWidth(labelFrame) + xOffset, CGRectGetHeight(labelFrame) } }];
+        UIView *titleMarginWrapper = [[UIView alloc] initWithFrame:(CGRect){CGPointZero, {CGRectGetWidth(labelFrame) + xOffset, CGRectGetHeight(labelFrame)}}];
```
Comment (alloy) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: Boooooo :stuck_out_tongue:

File: Artsy/View_Controllers/ARSerifNavigationViewController.m (hunk 1 above)
Comment (alloy) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: I’m not sure which property @ashfurrow means, but there is definitely a `shadowImage` property on `UINavigationBar`, which seems to be for this purpose.

This code below seems really fragile and hard to understand in what it does.

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift [hunk 3]
```diff
@@ -161,11 +155,11 @@ extension AuctionInformationViewController {
             
             self.stackView.constrainTopSpaceToView(self.flk_topLayoutGuide(), predicate: "0")
             self.stackView.alignLeading("0", trailing: "0", toView: self.view)
-            self.stackView.constrainBottomSpaceToView(self.flk_bottomLayoutGuide(), predicate: "0")
+            self.stackView.constrainBottomSpaceToView(self.flk_bottomLayoutGuide(), predicate: "-40")
```
Comment (alloy) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: What is this for?

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift [hunk 4]
```diff
@@ -205,10 +204,16 @@ extension AuctionInformationViewController {
...
                 titleLabel.font = UIFont.sansSerifFontWithSize(12)
+
+                let arrowIndicator = UIImageView(image: UIImage(named:"navigation_more_arrow_vertical"))
+                self.tapDirection = arrowIndicator
                 
                 let contentView = ARTextView()
                 contentView.scrollEnabled = true
-                contentView.setMarkdownString(entry.content)
+                /// The API comes with headers we should remove
+                let content = entry.content.stringByRemovingLinesMatching { $0.hasPrefix("#") }
```
Comment (alloy) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: I really don’t like that we 1) have to do this and 2) are re-implementing this on multiple clients.

Seeing as your in the office, could you figure out why this is being done at all? https://github.com/artsy/eigen/pull/1154#issuecomment-184636515

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift [hunk 5]
```diff
@@ -216,30 +221,36 @@ extension AuctionInformationViewController {
...
+                addSubview(titleButton)
                 titleButton.addSubview(titleLabel)
-                self.addSubview(contentView)
+                titleButton.addSubview(arrowIndicator)
+                addSubview(contentView)
                 
                 topBorder.constrainHeight("1")
                 topBorder.alignTopEdgeWithView(self, predicate: "0")
-                topBorder.alignLeading("0", trailing: "0", toView: self)
+                topBorder.alignLeading("20", trailing: "-20", toView: self)
```
Comment (alloy) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: Did the design on this change?

File: Artsy/Views/Auction/AuctionTitleView.swift [hunk 6]
```diff
@@ -10,9 +10,11 @@ import FLKAutoLayout
 }
 
 class AuctionTitleView: UIView {
-    weak var delegate: AuctionTitleViewDelegate!
+    weak var delegate: AuctionTitleViewDelegate?
 
     let viewModel: SaleViewModel
+    let showAdditionalInformation: Bool
```
Comment (alloy) [inline Artsy/Views/Auction/AuctionTitleView.swift]: Cool. This also means you can remove the optional parts of the protocol.

**Comment by alloy** [comment]
Looking forward to the updated snapshots for all those design changes.

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift (hunk 5 above)
Comment (orta) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: Ish - this is going to turn into a generic component - so I need to take it out of the Auctions context today and add a bool for toggling between these `0` & `20` - https://github.com/artsy/eigen/issues/1173

File: Artsy/Views/Auction/AuctionTitleView.swift (hunk 6 above)
Comment (orta) [inline Artsy/Views/Auction/AuctionTitleView.swift]: :+1:

**Comment by orta** [comment]
Yeah, there's a little bit more, figured it's not worth churning on a lot of snapshots with `ARNavigationButton`s

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift (hunk 5 above)
Comment (alloy) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: Components should not define their own margins imo, that should be left up to parent. I.e. in this case leave it 0 and if it needs to be 20 elsewhere then just apply that to the component?

File: Artsy/View_Controllers/ARSerifNavigationViewController.m (hunk 1 above)
Comment (orta) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: New to me, I just migrated code from folio where I've had to do this for many many years, nice there's a real API for it now

File: Artsy/View_Controllers/ARSerifNavigationViewController.m (hunk 2 above)
Comment (orta) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: :space_invader:

Comment (alloy) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: The computers are winning :disappointed:

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift (hunk 5 above)
Comment (orta) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: the title / image padding need to change too though

screen shot 2016-02-18 at 07 30 55

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift (hunk 3 above)
Comment (orta) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: I got confused with https://github.com/artsy/eigen/issues/1163 - will need to undo

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift (hunk 4 above)
Comment (orta) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: it's to give pages more flexibility, for example our past privacy policy has a different URL,  and shows a warning before the key title: https://admin.artsy.net/page/past-privacy-8-5-13

**Comment by orta** [comment]
OK, this is ready. I want to go through for one more iteration on checking if there's anything I added that should get extra tests, but it's good for review. 

@katarinabatina - you wanted to know about the changes to the navigation buttons - see the "File Changed" tab at the top and scroll to about half-way.

File: Artsy/View_Controllers/ARSerifNavigationViewController.m [hunk 7]
```diff
@@ -143,17 +144,17 @@ - (BOOL)definesPresentationContext
 
 - (UIModalPresentationStyle)modalPresentationStyle
 {
-    return self.shouldShowInForm ? UIModalPresentationFormSheet : UIModalPresentationFullScreen;
+    return UIModalPresentationFormSheet;
 }
 
-- (UIApplication *)sharedApp
+- (UIApplication *)sharedApplication
```
Comment (orta) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: @ashfurrow I bet _this_ was what fixed the snapshots

File: Artsy/View_Controllers/ARSerifNavigationViewController.m [hunk 8]
```diff
@@ -143,17 +144,17 @@ - (BOOL)definesPresentationContext
 
 - (UIModalPresentationStyle)modalPresentationStyle
 {
-    return self.shouldShowInForm ? UIModalPresentationFormSheet : UIModalPresentationFullScreen;
+    return UIModalPresentationFormSheet;
```
Comment (orta) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: this is full screen on iPhone, form on iPad

File: Artsy/View_Controllers/ARSerifNavigationViewController.m [hunk 9]
```diff
@@ -212,4 +213,11 @@ - (void)center:(UIView *)viewToCenter
     viewToCenter.frame = newFrame;
 }
 
+- (void)hideNavigationBarShadow:(BOOL)hide
+{
+    UIColor *color = hide ? [UIColor whiteColor] : [UIColor artsyMediumGrey];
+    [self setBackgroundImage:[UIImage imageFromColor:[UIColor whiteColor]] forBarMetrics:UIBarMetricsDefault];
```
Comment (orta) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: had to include this, or shadow image can't be changed

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift [hunk 10]
```diff
@@ -1,47 +1,46 @@
...
-    let startsAt: String
-    let contact: MarkdownString // TODO markdown ?
-    let FAQEntries: [FAQEntry]
-    
     struct FAQEntry {
         let name: String
-        let content: MarkdownString
+        let slug: String
+
+        let markdownSignal = Signal<MarkdownString>()
```
Comment (orta) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: using this effectively as a promise

**Comment by orta** [comment]
:green_apple:

File: Artsy/View_Controllers/ARSerifNavigationViewController.m (hunk 7 above)
Comment (ashfurrow) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: Huh, neat!

File: Artsy/View_Controllers/Auction/AuctionInformationViewController.swift (hunk 10 above)
Comment (ashfurrow) [inline Artsy/View_Controllers/Auction/AuctionInformationViewController.swift]: :+1:

File: Artsy/View_Controllers/Auction/AuctionViewController.swift [hunk 11]
```diff
@@ -187,13 +188,26 @@ extension AuctionViewController {
...
+                let registrationPath = "/auction-registration/\(self.saleID)"
+                let viewController = ARSwitchBoard.sharedInstance().loadPath(registrationPath)
+                self.navigationController?.pushViewController(viewController, animated: true)
+            }
+        }
+        if let _ = presentedViewController {
+            dismissViewControllerAnimated(true, completion: showRegister)
+        } else {
+            showRegister()
+        }
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Auction/AuctionViewController.swift]: Nice :ok_hand:

**Comment by ashfurrow** [comment]
Great!

**Pull request merged**